### PR TITLE
Sbiswas96/dispanno

### PR DESCRIPTION
**Pull request by sbiswas96**
@ldanzinger @anmacdonald please review the sample showing how to add Annotation Layer in Qt

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/DisplayAnnotation.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+class Map;
+class MapQuickView;
+}
+}
+
+#include <QObject>
+
+class DisplayAnnotation : public QObject
+{
+    Q_OBJECT
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/DisplayAnnotation.h]: Our coding standards are set to 2 spaces for tabs, as opposed to 4. This can be changed in the QtCreator IDE settings!

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/DisplayAnnotation.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#include "FeatureLayer.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "ServiceFeatureTable.h"
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+static const QUrl url("https://services1.arcgis.com/6677msI40mnLuuLr/arcgis/rest/services/East_Lothian_Rivers/FeatureServer/0");
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/DisplayAnnotation.cpp]: Be careful of the `static` keyword, it has different meanings in different contexts - some of which are C legacy! 

In this context the static keyword limits the value to file scope - exactly the same as an anonymous namespace, making the static keyword redundant!

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    "category": "TODO",
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json]: Looks like the se TODOs need filled in.

Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json]: Looks like these TODOs need filled in.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+            AnnotationLayer {
+                ServiceFeatureTable {
+                    id: annotationFeatureTable
+                    url: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/RiversAnnotation/FeatureServer/0"
+                }
+            }
+
+            initialViewpoint:
+                ViewpointCenter {
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: If you look at the ArcGIS Qt QML docs, a type of `ViewpointCenter` is automatically bound to `initialViewpoint` if declared within `AnnotationLayer`. Remove the `initialViewpoint:` line and fix the tabbing.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                // Specify the center Point
+                center: Point {
+                    x: -2.725610
+                    y: 55.882436
+                    spatialReference: SpatialReference { wkid: 4326 }
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: `spatialReference` can be removed.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                ServiceFeatureTable {
+                    id: annotationFeatureTable
+                    url: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/RiversAnnotation/FeatureServer/0"
+                }
+            }
+
+            initialViewpoint:
+                ViewpointCenter {
+                // Specify the center Point
+                center: Point {
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: `center` can be removed.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            BasemapLightGrayCanvas {}
+
+            FeatureLayer {
+                ServiceFeatureTable {
+                    id: featureTable
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This `id` is not required.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import Esri.ArcGISRuntime 100.8
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    MapView {
+        id: mapView
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This `id` is not required.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            FeatureLayer {
+                ServiceFeatureTable {
+                    id: featureTable
+                    url: "https://services1.arcgis.com/6677msI40mnLuuLr/arcgis/rest/services/East_Lothian_Rivers/FeatureServer/0"
+                }
+            }
+
+            AnnotationLayer {
+                ServiceFeatureTable {
+                    id: annotationFeatureTable
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This `id` is not required.

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
+# Display Annotation
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: should be sentence case

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+Display annotation from a feature service URL.
+![](screenshot.png)
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: should be a line return between description and image.

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+3. Add both layers to the operational layers of the map and add it to a `MapView`.
+
+## Relevant API
+ - AnnotationLayer
+ - FeatureLayer
+ - ServiceFeatureTable
+
+## About the data
+
+Data derived from [Ordnance Survey OpenRivers](https://www.ordnancesurvey.co.uk/business-government/products/open-map-rivers). Contains OS data &copy; Crown copyright and database right 2018.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: is the `&copy;` supposed to be here? I'm unsure what it means

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json (hunk 3 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json]: Yep, the metadata.json file is what powers the samples on the [developer site](https://developers.arcgis.com/qt/latest/qml/sample-code/analyze-hotspots/). It is a lot of duplicate info from the readme

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md (hunk 10 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: this issue is throughout the PR so I won't comment on this any further

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md (hunk 11 above)
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: screenshot should be 500x500 pixels. We also generally don't include the full application window so that it is more platform agnostic in the image. There are some details about the screenshot in the wiki I sent you on Slack

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+            FeatureLayer {
+                ServiceFeatureTable {
+                    id: featureTable
+                    url: "https://services1.arcgis.com/6677msI40mnLuuLr/arcgis/rest/services/East_Lothian_Rivers/FeatureServer/0"
+                }
+            }
+
+            AnnotationLayer {
+                ServiceFeatureTable {
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This code differs from the C++ implementation. C++ constructs via URL and QML constructs via a ServiceFeatureTable. We should match whatever the other SDKs are doing. Looks like on Android, the feature layer is constructed with a feature table and the annotation layer is constructed with a URL - https://github.com/Esri/arcgis-runtime-samples-android/blob/master/kotlin/display-annotation/src/main/java/com/esri/arcgisruntime/sample/displayannotation/MainActivity.kt#L38-L43

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml (hunk 4 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This is using a concept in QML called default properties - https://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html#default-properties

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.md [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+![](screenshot.png)
+
+## Use case
+Annotation is useful for displaying text that you don't want to move or resize when the map is panned or zoomed (unlike labels which will move and resize). You can use annotation to place text at a fixed size, position, orientation, font, and so on. You may choose to do this for cartographic reasons or because the exact placement of the text is important.
+
+## How to use the sample
+Pan and zoom to see names of waters and burns in a small region of Scotland.
+
+## How it works
+1. Create a `ServiceFeatureTable` with a URL.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.md]: These steps need to be updated once the QML code is changed. I think the same steps as C++ can be used. The only change is the wording - we don't usually say "constructed" in QML since there is no constructors. We just use the word "created" instead https://github.com/Esri/arcgis-runtime-samples-qt/pull/967/files#diff-a9e41a9d9c6a1f87da22c465352ae223R12

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md (hunk 12 above)
Comment (anmacdonald) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: Copyright symbol!

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.pro [hunk 15]
```diff
@@ -20,7 +20,7 @@ QT += opengl qml quick
 
 CONFIG += c++14
 
-ARCGIS_RUNTIME_VERSION = 100.8
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.pro]: This needs reverted back to 100.8

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml [hunk 16]
```diff
@@ -15,7 +15,7 @@
 // [Legal]
 
 import QtQuick 2.6
-import Esri.ArcGISRuntime 100.8
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: Revert back to 100.8

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md (hunk 12 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.md]: how come this renders properly in the QML readme but not the C++ readme?

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json [hunk 17]
```diff
@@ -0,0 +1,29 @@
+{
+    "category": "Layers",
+    "description": "Display annotation from a feature service URL.",
+    "ignore": false,
+    "images": [
+        "screenshot.png"
+    ],
+    "keywords": [
+        "AnnotationLayer",
+		"FeatureLayer",
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json]: looks like some tabs made their way in and messed up the format

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.json [hunk 18]
```diff
@@ -0,0 +1,5 @@
+{
+    "source" :  "DisplayAnnotation.qml",
+    "otherSourceFiles" :  [],
+    "name" : "Display Annotation"
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.json]: sentence case

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml (hunk 13 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: sounds like a bug. marking this resolved

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.metadata.json [hunk 19]
```diff
@@ -0,0 +1,27 @@
+{
+    "category": "Layers",
+    "description": "Display annotation from a feature service URL.",
+    "ignore": false,
+    "images": [
+        "screenshot.png"
+    ],
+    "keywords": [
+        "AnnotationLayer",
+		"FeatureLayer",
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.metadata.json]: tabs here as well

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+2. Create a `FeatureLayer` from the feature table and add it to the map.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/README.md]:
```suggestion
2. Create an `AnnotationLayer ` from the feature table and add it to the map.
```

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayWfsLayer/README.metadata.json [hunk 21]
```diff
@@ -9,15 +9,15 @@
...
-        "OGC",
-        "WFS",
-        "feature",
-        "web",
-        "service",
-        "layers",
-        "browse",
-        "catalog",
-        "interaction cache"
+		"OGC",
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayWfsLayer/README.metadata.json]: These changes should not be included in this PR. Also, we prefer spaces over tabs. Please remove this file from the PR

File: ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json (hunk 17 above)
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayAnnotation/README.metadata.json]: there are still lots of tab vs space issues in both metadata.json files. Please use spaces only

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml (hunk 13 above)
Comment (JamesMBallard) [reply ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayAnnotation/DisplayAnnotation.qml]: This will be fixed with 100.8 build 2702. I would recommend updating the sample to match the same workflow as C++ and other SDKs for consistency.
